cylc/flow/batch_sys_manager.py

```diff
@@ -634,6 +634,7 @@ def _job_submit_impl(
         # Submit job
         batch_sys = self._get_sys(batch_sys_name)
         proc_stdin_arg = None
+        # TODO: close stdin
```

Comment:
Can do this after line 678?

Reply:
Actually I had done it. I think I managed to eliminate the TODOs, except for one... then I reverted for not being brave enough to change, and not knowing how exactly to test it and make sure it didn't break anything.

Not entirely sure what code in `batch_sys_manager.py` is tested in Travis, and what part needs to invoke something like PBS/etc? But might be worth and check coverage in Codecov. Changed! :+1:

Reply:
~Changed~, scratch that. Still not confident in changing this :confused: 

If you look at the `if` statement right after this `TODO`, if I am reading it right, if the `batch_sys` has the attribute `get_submit_stdin`, then it looks like the value of `proc_stdin_value` is changed... so we may not have a resource in 678? And may instead have an `str` I guess?

Reply:
Sorry, looks like `.get_submit_stdin` is not documented properly. A quick `git grep` suggests that it is used by the `at` and `lsf` handlers.

Looking at the returned pair `proc_stdin_arg, proc_stdin_value`. The  `proc_stdin_arg` should be an opened file handle, `subprocess.PIPE` or `None`. (Not sure why it is tested for `str`.) The `proc_stdin_value` is either `None` or a `str` containing the content to pipe to STDIN.

Reply:
Huh, confusing. I think I will probably leave the TODO for now (if you would like to take over this PR and add/edit commits, feel free to push any changes here or start a new one)

Reply:
OK. I'll raise a PR against your branch.

Reply:
kinow/cylc-flow#4